autoendpoint/src/routers/fcm/client.rs

```diff
@@ -238,10 +238,17 @@ impl FcmClient {
         // Handle error
         let status = response.status();
         if status.is_client_error() || status.is_server_error() {
-            let data: FcmResponse = response
-                .json()
+            let raw_data = response
+                .bytes()
                 .await
                 .map_err(FcmError::DeserializeResponse)?;
+            if raw_data.is_empty() {
+                return Err(FcmError::EmptyResponse.into());
```

Comment:
The response `status` code would be nice to know. I think you could put it in both of these new error variants and include it in their to_string (for now, since we can't easily put it in `extras`) because I doubt we'll be seeing more than 1 status code anyway.